### DISCUSSION TITLE
How to convert JPG/PNG to multi resolution ico?

### DISCUSSION THREAD

**Discussion by sahilkumarmaths** [dlemstra/magick-wasm#74] [Q&A]
How to convert JPG/PNG image to multi resolution ico using magick-wasm?
I found the following that uses ImageMagick cli: `convert input.png -define icon:auto-resize="256,128,64,32,16" result.ico `

Moreover how to pass the options similar to imagemagick cli while using magick-wasm
Example: `magick -size 320x85 canvas:none -font Bookman-DemiItalic -pointsize 72 \
  -draw "text 25,60 \'Magick\'" -channel RGBA -blur 0x6 -fill darkred -stroke magenta \
  -draw "text 20,55 \'Magick\'" fuzzy-magick.png`

Thanks in advance :)

**Comment by dlemstra** [comment]
You will need to translate the command line code to typescript/javascript. It might be possible that a feature that you want is not yet available and should be added.

What have you tried yourself so far and where are you stuck?

**Reply by sahilkumarmaths** [reply]
Tried the following. The code creates an icon from image. Not sure how to create a multi resolution icon.

```
ImageMagick.read(imageData, (image) => {
	image.resize(100, 100);
	image.write((data) => {
		console.log(data);
	}, MagickFormat.Ico);
});
```

```
magick identify .\icon.ico
.\icon.ico ICO 99x100 99x100+0+0 8-bit sRGB 41262B 0.000u 0:00.001
```

**Reply by dlemstra** [reply]
With the current release it is not possible to set the `icon:auto-resize` define but you will be able to do this in the next release:

```js
ImageMagick.read(imageData, (image) => {
    image.resize(100, 100);
    image.settings.setDefine('icon:auto-resize', '64,32,16'); // removed 256 and 128 because of resize to 100x100.
    image.write((data) => {
        console.log(data);
    }, MagickFormat.Ico);
});
```
